File: packages/typescript/src/resources/interactions.ts
/**
 * Interactions resource — single-trace lookup by interaction id.
 *
 * The agent-friendly entry point for "explain this trace" and "generate
 * a waterfall for this interaction" prompts.
 */

import { ConvaiAnalyticsError } from "../apiErrors.js";
import type { ConvaiAnalytics } from "../client.js";
import type { InteractionTrace } from "../types.js";

export class Interactions {
  readonly #client: ConvaiAnalytics;

  constructor(client: ConvaiAnalytics) {
    this.#client = client;
  }

  /**
   * `GET /v1/analytics/interactions/{id}` — full component waterfall +
   * provider/model attribution + status for one interaction.
   */
  async get(interactionId: string): Promise<InteractionTrace> {
    if (!interactionId) {
      throw new ConvaiAnalyticsError(0, {
        code: "missing_argument",
        message: "interactions.get(interactionId): interactionId is required",
      });
    }
    return this.#client.get<InteractionTrace>(
      `/interactions/${encodeURIComponent(interactionId)}`,
      {},
    );
  }
}

File: packages/python/convai_analytics/resources/interactions.py
"""Interactions resource — single-trace lookup by interaction id.

The agent-friendly entry point for "explain this trace" and
"generate a waterfall for this interaction" prompts.
"""

from __future__ import annotations

from typing import TYPE_CHECKING

from ..errors import ConvaiAnalyticsError
from ..types import InteractionTrace

if TYPE_CHECKING:
    from ..client import ConvaiAnalytics


class Interactions:
    def __init__(self, client: ConvaiAnalytics) -> None:
        self._client = client

    def get(self, interaction_id: str) -> InteractionTrace:
        """``GET /v1/analytics/interactions/{id}`` — full component waterfall."""
        if not interaction_id:
            raise ConvaiAnalyticsError(
                0,
                "missing_argument",
                "interactions.get(interaction_id): interaction_id is required",
            )
        return InteractionTrace.model_validate(
            self._client._get(f"/interactions/{interaction_id}", {}),
        )
